code/datums/status_effects/buffs/stop_drop_roll.dm

```diff
@@ -0,0 +1,66 @@
+/datum/status_effect/stop_drop_roll
+	id = "stop_drop_roll"
+	alert_type = null
+
+	tick_interval = 0.8 SECONDS
+
+/datum/status_effect/stop_drop_roll/on_apply()
+	if(!iscarbon(owner))
+		return FALSE
+
+	var/actual_interval = initial(tick_interval)
+	if(!owner.Knockdown(actual_interval * 2, ignore_canstun = TRUE) || owner.body_position != LYING_DOWN)
+		to_chat(owner, span_warning("You try to stop, drop, and roll - but you can't get on the ground!"))
+		return FALSE
+
+	RegisterSignal(owner, COMSIG_MOVABLE_MOVED, PROC_REF(stop_rolling))
+	RegisterSignal(owner, COMSIG_LIVING_SET_BODY_POSITION, PROC_REF(body_position_changed))
+
+	owner.visible_message(
+		span_danger("[owner] rolls on the floor, trying to put [owner.p_them()]self out!"),
+		span_notice("You stop, drop, and roll!"),
+	)
+	// Start with one weaker roll
+	owner.spin(spintime = actual_interval, speed = actual_interval / 4)
+	owner.adjust_fire_stacks(-0.5)
+	ADD_TRAIT(owner, TRAIT_HANDS_BLOCKED, id) // they're busy!
```

Comment:
use `TRAIT_STATUS_EFFECT()` here?

Reply:
I opted not to so it'd just show up as `stop_drop_roll` in vv (for clarity), but if you think we'll add another stop drop and roll trait source down the line i'll change it

Reply:
Ultimately meh then, just feels awk